adapters/dxtech/dxtech.go

```diff
@@ -0,0 +1,173 @@
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		request.Imp = []openrtb2.Imp{impression}
+		body, err := json.Marshal(request)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+
+		if request.Test == 1 {
+			impExt.PublisherId = "test"
+		}
+
+		params := url.Values{}
+		params.Add("publisher_id", impExt.PublisherId)
+		params.Add("placement_id", impExt.PlacementId)
+
+		adapterRequests = append(adapterRequests, &adapters.RequestData{
+			Method:  http.MethodPost,
+			Uri:     a.endpoint + "?" + params.Encode(),
+			Body:    body,
+			Headers: getHeaders(request),
+			ImpIDs:  openrtb_ext.GetImpIDs(request.Imp),
+		})
+	}
+
+	request.Imp = impressions
+	return adapterRequests, errs
+}
+
+func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(response) {
+		return nil, nil
+	}
+	if err := adapters.CheckResponseStatusCodeForErrors(response); err != nil {
+		return nil, []error{err}
+	}
+
+	var ortbResponse openrtb2.BidResponse
+	err := jsonutil.Unmarshal(response.Body, &ortbResponse)
+	if err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: "Bad Server Response",
+		}}
+	}
+
+	var bidErrors []error
+
+	bidderResponse := adapters.NewBidderResponseWithBidsCapacity(1)
+	for _, seatBid := range ortbResponse.SeatBid {
+		for i := range seatBid.Bid {
+			bid := seatBid.Bid[i]
+			bidType, err := getBidType(&bid)
+			if err != nil {
+				bidErrors = append(bidErrors, err)
+				continue
+			}
+
+			bidderResponse.Bids = append(bidderResponse.Bids, &adapters.TypedBid{
+				Bid:     &bid,
+				BidType: bidType,
+			})
+		}
+	}
+
+	return bidderResponse, bidErrors
+}
+
+func getBidType(bid *openrtb2.Bid) (openrtb_ext.BidType, error) {
+	if bidType, ok := markupTypeToBidType[bid.MType]; ok {
+		return bidType, nil
+	}
+	return "", &errortypes.BadServerResponse{
+		Message: fmt.Sprintf("Unsupported MType %d", bid.MType),
+	}
+}
+
+func parseExt(imp *openrtb2.Imp) (*openrtb_ext.ExtImpDXTech, error) {
+	var bidderExt adapters.ExtImpBidder
+
+	if err := jsonutil.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return nil, &errortypes.BadInput{
+			Message: fmt.Sprintf("Ignoring imp id=%s, error while decoding extImpBidder, err: %s", imp.ID, err),
+		}
+	}
+
+	impExt := openrtb_ext.ExtImpDXTech{}
+	err := jsonutil.Unmarshal(bidderExt.Bidder, &impExt)
+	if err != nil {
+		return nil, &errortypes.BadInput{
+			Message: fmt.Sprintf("Ignoring imp id=%s, error while decoding impExt, err: %s", imp.ID, err),
+		}
+	}
+
+	return &impExt, nil
+}
+
+func getHeaders(request *openrtb2.BidRequest) http.Header {
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("X-Openrtb-Version", "2.5")
+
+	if request.Site != nil {

```

Comment:
Just to double-check: your adapter also supports `app` according to your `static/bidder-info/dxtech.yaml` file. If a given request comes with `request.App` and not `request.Site`, should any header, like maybe `Origin`, be added?